fairscale/nn/data_parallel/fsdp_optim_utils.py

```diff
@@ -55,7 +58,10 @@ def flatten_optim_state_dict(sd: Dict) -> Dict:
 
     # add pointers from the `params` dict.
     for pg_id, _ in enumerate(sd["param_groups"]):
-        # TODO: this list could be huge. Can we avoid materializing?
+        # The values() list may look like [0,0,None,None,2,2]. We use
```

Comment:
Will the values() list have None if you have removed it in line 26 above?

Reply:
line 26 above removed None from a set we computed from the value() call. Value() call here still returns the same list that may have None.